### PR TITLE
fix(Stored): Make stores typed

### PR DESCRIPTION
**Pull request by Seitseman**
adding types instead of var for stores used in qml components
fixes: #14801

### CODE REVIEW

**Comment by alexjba** [comment]
@Seitseman Try a rebase to latest master

**Comment by alexjba** [comment]
The UI tests are failing with a legit error

```
[2024-06-03T13:02:35.931Z] 3: PASS   : QmlTests::StatusAmountInput::cleanupTestCase()
[2024-06-03T13:02:35.931Z] 3: QWARN  : QmlTests::UnknownTestFunc() file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15011/storybook/qmlTests/tests/tst_StatusChatInput.qml:21:9: Type StatusChatInput unavailable 
[2024-06-03T13:02:35.931Z] 3:              StatusChatInput { 
[2024-06-03T13:02:35.931Z] 3:              ^
[2024-06-03T13:02:35.931Z] 3: QWARN  : QmlTests::UnknownTestFunc() file:///home/jenkins/workspace/s_linux_x86_64_tests-ui_PR-15011/ui/imports/shared/status/StatusChatInput.qml:39:5: ChatStores.UsersStore is not a type 
[2024-06-03T13:02:35.931Z] 3:          property ChatStores.UsersStore usersStore 
[2024-06-03T13:02:35.931Z] 3:          ^
```

**Review by micieslak — COMMENTED** [review]
Massive work! It's great initial step for further improvements! Looks good in general but there are some stubs missing causing qml tests and pages validator failing. Adding stubs should quickly help.

**Review by micieslak — APPROVED** [review]
Good job!

File: ui/app/AppLayouts/Chat/views/ChatHeaderContentView.qml [hunk 1]
```diff
@@ -21,7 +21,7 @@ Item {
     property alias membersButton: membersButton
     property alias searchButton: searchButton
 
-    property var rootStore
+    required property RootStore rootStore
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/ChatHeaderContentView.qml:24]: `required` added here intentionally?

File: ui/app/AppLayouts/Chat/views/ChatMessagesView.qml [hunk 2]
```diff
@@ -18,17 +18,20 @@ import shared.status 1.0
 import shared.controls 1.0
 import shared.views.chat 1.0
 
+import AppLayouts.Profile.stores 1.0
+
 import "../controls"
 import "../panels"
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/ChatMessagesView.qml:25]: Needed? Given that you also added the typed import above

File: ui/app/AppLayouts/Chat/views/ContactsColumnView.qml [hunk 3]
```diff
@@ -19,7 +19,9 @@ import SortFilterProxyModel 0.2
 
 import "../panels"
 import "../popups"
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/ContactsColumnView.qml:22]: Needed?

File: ui/app/AppLayouts/Profile/views/EnsDetailsView.qml [hunk 4]
```diff
@@ -13,10 +13,12 @@ import shared.popups 1.0
 import shared.popups.send 1.0
 import shared.stores.send 1.0
 
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/EnsDetailsView.qml:16]: Better a typed import?

File: ui/app/AppLayouts/Profile/views/EnsSearchView.qml [hunk 5]
```diff
@@ -15,11 +15,13 @@ import shared.controls 1.0
 import shared.popups.send 1.0
 import shared.stores.send 1.0
 
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/EnsSearchView.qml:18]: Better a typed import?

File: ui/app/AppLayouts/Profile/views/EnsTermsAndConditionsView.qml [hunk 6]
```diff
@@ -18,11 +18,13 @@ import StatusQ.Components 0.1
 
 import AppLayouts.Wallet.stores 1.0
 
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/EnsTermsAndConditionsView.qml:21]: Typed import

File: ui/app/AppLayouts/Profile/views/LeftTabView.qml [hunk 7]
```diff
@@ -9,11 +9,12 @@ import shared 1.0
 import shared.popups 1.0
 
 import "../panels"
+import "../stores"
```
Comment (caybro) [inline ui/app/AppLayouts/Profile/views/LeftTabView.qml:12]: Dtto

File: ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml [hunk 8]
```diff
@@ -20,7 +20,7 @@ import SortFilterProxyModel 0.2
 
 import AppLayouts.stores 1.0
 
-import "../stores"
+import "../stores" as WalletStores
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/popups/AddEditSavedAddressPopup.qml:23]: Typed import?

File: ui/app/mainui/Popups.qml [hunk 9]
```diff
@@ -255,7 +258,9 @@ QtObject {
         }
     }
 
-    function openPinnedMessagesPopup(store, messageStore, pinnedMessagesModel, messageToPin, chatId) {
+    function openPinnedMessagesPopup(store: ChatStores.RootStore,
```
Comment (caybro) [inline ui/app/mainui/Popups.qml:261]: This looks definitely wrong

File: ui/app/AppLayouts/Chat/views/ChatHeaderContentView.qml (hunk 1 above)
Comment (Seitseman) [reply ui/app/AppLayouts/Chat/views/ChatHeaderContentView.qml:24]: To be removed, not needed here

File: ui/app/AppLayouts/Chat/views/ChatMessagesView.qml (hunk 2 above)
Comment (Seitseman) [reply ui/app/AppLayouts/Chat/views/ChatMessagesView.qml:25]: Different store, "Profile" vs "Chat".  I'll import it as module, not directory.

**Review by alexjba — APPROVED** [review]
Awesome work! LGTM!

**Comment by noeliaSD** [comment]
As agreed offline @Seitseman, I changed it to _draft_ and we will integrate it during the stability weeks to reduce the impact of possible regressions for the next release **2.30**.

In any case, good job!

File: ui/app/AppLayouts/Profile/views/AboutView.qml [hunk 10]
```diff
@@ -15,7 +15,7 @@ import shared.status 1.0
 SettingsContentBase {
     id: root
 
-    property var store
```
[the remark this answers is not in the document]
Comment (micieslak) [reply ui/app/AppLayouts/Profile/views/AboutView.qml:18]: It turns out that in `ProfileLayout.qml` the store is defined in-place as `QtObject` for some reason :man_shrugging: There is no dedicated, typed store defined.

**Pull request merged**